### PR TITLE
Add HITL test for `SYNC_IN`/`SYNC_OUT` hardware

### PR DESCRIPTION
**Pull request by martijnbastiaan**
(Assigning multiple people as reviewer, but no need to do 3 reviews :))

### CODE REVIEW

File: bittide-instances/data/constraints/syncInSyncOut.xdc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# White   | PMOD0_4       | IO5           | SYNC_IN
+# Purple  | PMOD0_5       | IO6           | FDEC
+# Green   | PMOD0_6       | IO7           | CSB
+# Orange  | PMOD0_7       | IO8           | MISO/SDO
+# Black   | Not connected | Not connected |
+# Brown   | PMOD_GND      | GND           | GND (SPI)
+
+# PMOD1_[0..7]
+set_property -dict {IOSTANDARD LVCMOS12 PACKAGE_PIN AK25} [get_ports {SYNC_OUT}]
+set_property -dict {IOSTANDARD LVCMOS12 PACKAGE_PIN AE26} [get_ports {SYNC_IN}]
```
Comment (hiddemoll) [inline bittide-instances/data/constraints/syncInSyncOut.xdc:23]:
```suggestion
# PMOD1_0
set_property -dict {IOSTANDARD LVCMOS12 PACKAGE_PIN AK25} [get_ports {SYNC_OUT}]
# PMOD1_4
set_property -dict {IOSTANDARD LVCMOS12 PACKAGE_PIN AE26} [get_ports {SYNC_IN}]
```

File: bittide-instances/src/Bittide/Instances/Tests/SyncInSyncOut.hs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import Bittide.Instances.Domains
+import Bittide.Arithmetic.Time
+
+import Clash.Annotations.TH
+import Clash.Cores.Xilinx.VIO
+import Clash.Cores.Xilinx.Xpm.Cdc.Single
+import Clash.Reset.Extra
+import Clash.Xilinx.ClockGen
+
+-- | An 'Index' countin to /n/ seconds on 'Basic300'
```
Comment (hiddemoll) [inline bittide-instances/src/Bittide/Instances/Tests/SyncInSyncOut.hs:74]:
```suggestion
-- | An 'Index' counting to /n/ seconds on 'Basic300'
```

File: bittide-shake/bin/Shake.hs [hunk 3]
```diff
@@ -192,6 +192,7 @@ targets = map enforceValidTarget
   , testTarget "Bittide.Instances.BoardTest.simpleHardwareInTheLoopTest"
   , testTarget "Bittide.Instances.Tests.FincFdec.fincFdecTests"
   , testTarget "Bittide.Instances.Tests.VexRiscv.vexRiscvTest"
+  , testTarget "Bittide.Instances.Tests.SyncInSyncOut.syncInSyncOut"
```
Comment (hiddemoll) [inline bittide-shake/bin/Shake.hs:195]: Can you sort these targets?

File: bittide-instances/src/Bittide/Instances/Tests/SyncInSyncOut.hs [hunk 4]
```diff
@@ -0,0 +1,164 @@
+-- SPDX-FileCopyrightText: 2022-2023 Google LLC
```
Comment (hiddemoll) [inline bittide-instances/src/Bittide/Instances/Tests/SyncInSyncOut.hs:1]:
```suggestion
-- SPDX-FileCopyrightText: 2023 Google LLC
```

**Comment by martijnbastiaan** [comment]
@hiddemoll Test results:

* `Specific [7,6,5,4,3,2,1,0])`: https://github.com/bittide/bittide-hardware/actions/runs/5783856561/job/15674616368
* `Specific [0,1,2,3,4,5,7,6])`: https://github.com/bittide/bittide-hardware/actions/runs/5784287852/job/15675222642

**Pull request merged**